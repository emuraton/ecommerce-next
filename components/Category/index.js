import React from 'react';
import styled from 'styled-components';

const Button = styled.button`
  position: relative;
  display: inline-block;
  list-style: none;
  padding: 12px 10px;
  border-radius: 25px;
  /* background-color: rgba(36, 37, 41, 0.05); */
  background-color: white;
  color: rgb(36, 37, 41);
  border-color: rgb(162, 164, 173);
  text-align: center;
  text-decoration: none;
  outline: none;
`;

const Label = styled.span`
  font-size: 16px;
  font-weight: 500;
  padding: 10px 15px;
`;

const Category = ({ label }) => {
  return (
    <div>
      <Button>
        <Label>{label}</Label>
      </Button>
    </div>
  );
};

export default Category;
