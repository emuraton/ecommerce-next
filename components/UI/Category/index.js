import React from 'react';
import styled, { css } from 'styled-components';
import { ifProp } from 'styled-tools';

const Button = styled.button`
  position: relative;
  display: inline-block;
  list-style: none;
  padding: 12px 10px;
  border-radius: 25px;
  background-color: white;
  color: rgb(36, 37, 41);
  border: 1px solid rgb(162, 164, 173);
  text-align: center;
  outline: none;
  cursor: pointer;

  ${ifProp(
    'active',
    css`
      background-color: rgb(0, 195, 239);
      color: white;
    `,
  )};
`;
Button.displayName = 'Button';

const Label = styled.span`
  font-size: 16px;
  font-weight: 500;
  padding: 10px 15px;
`;

const Category = ({ slug, name, active, onClick }) => {
  return (
    <div>
      <Button type="button" name={slug} active={active} onClick={onClick}>
        <Label>{name}</Label>
      </Button>
    </div>
  );
};

export default Category;
