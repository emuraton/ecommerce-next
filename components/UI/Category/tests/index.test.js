import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';

import Category from '../';

describe('<Category />', () => {
  const initialProps = {
    slug: 'ticket',
    name: 'Ticket',
    active: false,
    onClick: Function.prototype,
  };
  describe('@render', () => {
    it('default render', () => {
      const wrapper = shallow(<Category {...initialProps} />);
      const button = wrapper.find('Button');
      expect(button).toHaveStyleRule('color', 'rgb(36,37,41)');
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('check css from active props', () => {
      const props = { ...initialProps, active: true };
      const wrapper = shallow(<Category {...props} />);
      const button = wrapper.find('Button');
      expect(button).toHaveStyleRule('color', 'white');
      expect(button).toHaveStyleRule('background-color', 'rgb(0,195,239)');
    });
  });
});
