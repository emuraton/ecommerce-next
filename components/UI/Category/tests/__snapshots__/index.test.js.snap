// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Category /> @render default render 1`] = `
<div>
  <Button
    active={false}
    name="ticket"
    onClick={[Function]}
  >
    <Category__Label>
      Ticket
    </Category__Label>
  </Button>
</div>
`;
